src/Voron/Data/VoronStream.cs

```diff
@@ -67,32 +75,26 @@ public override long Position
         {
             get
             {
-                long pos = 0;
-                // ReSharper disable once ForCanBeConvertedToForeach
-                for (int i = 0; i < _positions.Length; i++)
-                {
-                    pos += _positions[i];
-                }
-                return pos;
+                return _position;
             }
             set
             {
-                long pos = 0;
-                _index = _positions.Length - 1;
-                for (int i = 0; i < _positions.Length; i++)
+                var search = Array.BinarySearch(_chunksOffsets, value);
+                if (search >= 0)
                 {
-                    if (pos + _chunksDetails[i].ChunkSize > value)
-                    {
-                        _positions[i] = (int)((value - pos) % _chunksDetails[i].ChunkSize);
-                        _index = i;
-                        break;
-                    }
-                    _positions[i] = _chunksDetails[i].ChunkSize;
-                    pos += _chunksDetails[i].ChunkSize;
+                    //The index of the specified value in the specified array, if value is found; otherwise, a negative number
+                    _position = value;
+                    _index = search;
                 }
-                for (int i = _index + 1; i < _positions.Length; i++)
+                else
                 {
-                    _positions[i] = 0;
+                    //If value is not found and value is less than one or more elements in array, the negative number returned is the bitwise complement of the index of the first element that is larger than value.
+                    //If value is not found and value is greater than all elements in array, the negative number returned is the bitwise complement of (the index of the last element plus 1). If this method is called with a non-sorted array, the return value can be incorrect and a negative number could be returned, even if value is present in array.
+                    search = ~search;
+                    
+                    //TODO what to do when value is out of total length of chunks?
```

Comment:
That means that is is being set beyond the length of the file.
Set it to the end

Reply:
Ok, I will do that.